server/cluster/coordinator.go

```diff
@@ -155,6 +160,45 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+// checkMissRegions check miss peer, if miss region is more than maxMissPeerQueueSize,
+// it will return ture to avoid run other replica check
+func (c *coordinator) checkMissRegions() bool {
+	updates := make([]*cache.Entry, 0)
+	removes := make([]uint64, 0)
+	missPeers := c.checkers.GetMissRegions()
+	regionListGauge.WithLabelValues("priority").Set(float64(len(missPeers)))
+	log.Debug("check miss regions", zap.Int("miss region count", len(missPeers)))
+	for _, entry := range missPeers {
+		id, ok := entry.Value.(uint64)
+		if !ok {
+			continue
+		}
+		region := c.cluster.GetRegion(id)
+		if region == nil {
+			// the region could be recent split, continue to wait.
+			removes = append(removes, id)
+			continue
+		}
+		ops := c.checkers.CheckRegion(region)
+		// skip merge operator
+		if len(ops) == 0 || ops[0].Kind()&operator.OpMerge != 0 {
+			continue
+		}
+		if !c.opController.ExceedStoreLimit(ops...) {
+			c.opController.AddWaitingOperator(ops...)
+			if entry.Priority = entry.Priority - 1; entry.Priority > 1 {
+				updates = append(updates, entry)
```

Comment:
I think a better way to update the priority is when the operator finishes.

Reply:
it will update in check region